File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        if root == None:
            return []
        Q = [root]
        nextCount = 1
        while(len(Q) > 0):
            count = nextCount
            nextCount = 0
            currRes = []
            for i in range(count):
                currItem = Q.pop(0)
                currRes.append(currItem.val)
                if currItem.left != None:
                    Q.append(currItem.left)
                    nextCount+=1
                if currItem.right != None:
                    Q.append(currItem.right)
                    nextCount+=1
            res.append(currRes)
        return res

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> outerList = new ArrayList<>();
        if(root == null){
            return outerList;
        }
        Queue<TreeNode> queue = new LinkedList<>();
        queue.add(root);
        queue.add(null);
        while(queue.size() > 1){
            List<Integer> innerList = new ArrayList<>();
            while(queue.peek() != null){
                TreeNode i = queue.poll();
                innerList.add(i.val);
                if(i.left != null){
                    queue.add(i.left);
                }   
                if(i.right != null){
                    queue.add(i.right);
                }
            }
            queue.add(queue.poll());
            outerList.add(innerList);
        }
        return outerList;
    }
}